src/wasm/wasm-s-parser.cpp

```diff
@@ -1966,15 +1973,38 @@ Expression* SExpressionWasmBuilder::makeTry(Element& s) {
   }
   auto label = nameMapper.pushLabelName(sName);
   Type type = parseOptionalResultType(s, i); // signature
+
   if (!elementStartsWith(*s[i], "do")) {
     throw ParseException(
       "try body should start with 'do'", s[i]->line, s[i]->col);
   }
-  ret->body = makeTryOrCatchBody(*s[i++], type, true);
-  if (!elementStartsWith(*s[i], "catch")) {
-    throw ParseException("catch clause does not exist", s[i]->line, s[i]->col);
+  ret->body = makeMaybeBlock(*s[i++], 1, type);
+
+  while (i < s.size() && elementStartsWith(*s[i], "catch")) {
```

Comment:
Shouldn't this check for equality rather than for a prefix? Same below for catch_all.

Reply:
Not sure what you mean by the 'prefix'. Do you mean I should check if the string is exactly `catch` and not things like `catchblah`? [`elementStartsWith`](https://github.com/WebAssembly/binaryen/blob/fcbefa951e3e514f6b7555664b89fb786850f6bc/src/wasm/wasm-s-parser.cpp#L65-L67) does not check string prefixes; it checks element prefixes. For example, `elementStartsWith(expr, "catch")` returns true for any `Element` in the form of `(catch ...)`.

Reply:
Oops, I misunderstood what `elementStartsWith` was doing. Looks good!